EvalTables: Preview (#2892)
wandb/docs: models/evaltables/compare-runs.mdx

---
title: Compare runs with eval tables
---

> Use Eval Tables to compare inputs, outputs, and scores across multiple runs. W&B aligns corresponding examples and calculates aggregate scores and score differences for the selected runs.

Reviewer: Is it worth differentiating this from the compare-runs page / baseline and pinned runs?

To compare runs:

1. Navigate to your project workspace.
2. Scroll to the Evaluation Tables panel.
3. Select the **+** button.
4. Select the run that you want to add.

> W&B groups columns that have the same role and name across the selected runs. For example, if two runs use the same `input_columns`, W&B displays those columns together in the **Inputs** section. W&B similarly groups shared `output_columns` and `score_columns` in the **Outputs** and **Scores** sections.

Reviewer: ```suggestion
W&B groups columns that have the same role and name across the selected runs. For example, if two runs use the same `input_columns`, W&B displays those columns together in the **Inputs** section.  Shared `output_columns` and `score_columns` are also grouped in the **Outputs** and **Scores** sections.
```
Somehow, it feels clearer even though it's passive. Optional, though.

> The following image shows an Eval Table that compares the `summer-butterfly-9` and `gentle-flower-8` runs:

Reviewer: ```suggestion
The following screenshot shows an Eval Table that compares the `summer-butterfly-9` and `gentle-flower-8` runs:
```
More accessible

> <Frame>

Reviewer: Nit: I don't think you need to set the width

    <img src="/images/evaltables/overview_basic_compare_two_runs.png" alt="Evaluation table view" width="100%" />
</Frame>

## Set the reference run

> When you compare multiple runs, W&B designates one run as the reference run. W&B uses the reference run as the baseline when it calculates score deltas.

Reviewer: ```suggestion
When you compare multiple runs, W&B defaults to designating the left-most run as the reference run. The reference run is the the baseline to calculate score deltas.
```

> By default, the leftmost run is the reference run. To choose a different reference run, hover over the run, open its menu, and select **Set as reference**.

Reviewer: ```suggestion
To choose a different reference run, hover over the run, open its menu, and select **Set as reference**.
```

## View aggregate scores

W&B calculates an aggregate value for each score column in each selected run. The calculation depends on the score's data type:

| Type of score | Aggregate value |
| --- | --- |
| Boolean | Count and fraction of `true` values |
| Numeric | Mean |
| String or categorical | No scalar aggregate |
| Null | Excluded from calculation |

The following image highlights the aggregate values for the correct and confidence scores:

<Frame>
    <img src="/images/evaltables/overview_basic_compare_two_runs_labled.png" alt="Evaluation table view" width="100%" />
</Frame>

## Compare score deltas

For each score column, W&B calculates the difference between the reference run and every other selected run.

W&B calculates each delta as:

> comparison run value - reference run value

Reviewer: I think Mintlify supports ```math lexer

Author: I used text because math looked ruff. (It was highly italicized, odd space => almost unreadable)
